weld-rendering/app/src/main/java/styler/weld/rendering/factory/FactoryMethod.kt

```diff
@@ -0,0 +1,8 @@
+package styler.weld.rendering.factory
+
+import android.view.ViewGroup
+import styler.weld.rendering.viewholder.BaseViewHolder
+
+class FactoryMethod(val type: String, val factory: (parent: ViewGroup) -> BaseViewHolder) {
+    fun createViewHolder(parent: ViewGroup): BaseViewHolder = factory(parent)
+}
```

Comment:
I extracted this `FactoryMethod` in case there is another Factory that might use this class.